<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An R Shiny Application for annotating outbreak phylogenetic
  trees • tinselR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="An R Shiny Application for annotating outbreak phylogenetic
  trees">
<meta property="og:description" content="This is a phylogenetic tree application for adding
  additional information to trees. Additional information could come from a
  genetic distance file or meta data file.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">tinselR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/tinselR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jennahamlin/tinselR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="what-is-tinselr" class="section level3">
<h3 class="hasAnchor">
<a href="#what-is-tinselr" class="anchor"></a>What is tinselR?</h3>
<p>tinselR (pronunced tinsel-er) at its’ most basic level is a graphical viewer of Newick-formatted phylogenetic trees and as an application for producing publication-ready figures. The <strong>power</strong> of tinselR comes with combining a genetic distance matrix for annotating a tree for epidemiological outbreak analyses. A genetic distance matrix contains SNP differences for all pairwise comparisons for the tips on the tree. One can also include a <a href="https://yulab-smu.top/treedata-book/chapter7.html">heatmap</a> when that data is provided in the meta data file.</p>
</div>
<div id="issues-problems-suggestions-thoughts" class="section level3">
<h3 class="hasAnchor">
<a href="#issues-problems-suggestions-thoughts" class="anchor"></a>Issues, problems, suggestions, thoughts</h3>
<p>If you have any the above, please submit an <em>issue</em> on github located <a href="https://github.com/jennahamlin/tinselR/issues">here</a>.</p>
</div>
<div id="requires--" class="section level3">
<h3 class="hasAnchor">
<a href="#requires--" class="anchor"></a>Requires -</h3>
<ul>
<li>
<a href="https://bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a>; see the quick start for how to install ggtree.</li>
<li>
<a href="http://bioconductor.org/packages/release/bioc/html/treeio.html">treeio</a>; which will install with ggtree.</li>
<li>R version &gt;= 3.5</li>
</ul>
<!-- badges: start --><p><a href="https://travis-ci.org/jennahamlin/tinselR"><img src="https://travis-ci.org/jennahamlin/tinselR.svg?branch=master" alt="Travis build status"></a> <!-- badges: end --></p>
<hr>
</div>
<div id="quick-start" class="section level3">
<h3 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick Start</h3>
<p><strong>1). Install devtools package</strong></p>
<p>Run this code in your R console -</p>
<p><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("devtools", dep=T)</a></code></p>
<p><strong>2). Install ggtree and treeio</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2"></a>    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">"ggtree"</span>)</span></code></pre></div>
<p><strong>3). Install and launch the tinselR shiny application</strong></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">"jennahamlin/tinselR"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">library</span>(tinselR)</span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="kw">run_app</span>()</span></code></pre></div>
<p><strong>4). Load your data or use the example data</strong></p>
<p><em>Please click on the ‘Data Upload’ pane, where you can upload your files.</em> <em>Example data is available for playing with tinselR in the ‘Example Data’ pane</em> <em>of the application. The example data sets include all three files (tree,</em> <em>gene, and meta) and are available for selection via one drop down menu</em></p>
<ul>
<li><p><strong>Phylogenetic Tree</strong> - required; a <a href="https://en.wikipedia.org/wiki/Newick_format">Newick</a> generated tree</p></li>
<li><p><strong>Genetic Distance data</strong> - optional for use with the annotation function; a tsv/txt/csv file - see below for image of genetic distance matrix.</p></li>
</ul>
<p>
</p>
<h4>
genetic Distance screenshot
</h4>
<p><img src="reference/figures/geneDistanceExample.PNG"></p>

<ul>
<li>
<strong>Metadata</strong> - optional for easy correction of tip labels or adding a heatmap; a tsv/txt/csv file - requires column headers of Display.labels and Tip.labels. See image below for a csv file example.</li>
</ul>
<p>
</p>
<h4>
meta Data screenshot
</h4>
<p><img src="reference/figures/metaDataExample.PNG"></p>

<p><strong>5). Alter or annotate your tree</strong></p>
<div id="once-the-phylogenetic-tree-is-uploaded-you-can--" class="section level4">
<h4 class="hasAnchor">
<a href="#once-the-phylogenetic-tree-is-uploaded-you-can--" class="anchor"></a>Once the phylogenetic tree is uploaded you can -</h4>
<ul>
<li>Alter visualization parameters. See below for tree with aligned tips.</li>
</ul>
<p>
</p>
<p><img src="reference/figures/Slide4.PNG"></p>

</div>
<div id="once-the-genetic-distance-file-is-uploaded-you-can--" class="section level4">
<h4 class="hasAnchor">
<a href="#once-the-genetic-distance-file-is-uploaded-you-can--" class="anchor"></a>Once the genetic distance file is uploaded you can -</h4>
<ul>
<li>add annotation to the visual representation of the tree. See below for a tree with annotated clades including the range of SNPs.</li>
</ul>
<p>
</p>
<p><img src="reference/figures/Slide5.PNG"></p>

</div>
<div id="if-column-for-heatmap-is-included-in-metadata-file-you-can--" class="section level4">
<h4 class="hasAnchor">
<a href="#if-column-for-heatmap-is-included-in-metadata-file-you-can--" class="anchor"></a>If column for heatmap is included in metadata file you can -</h4>
<ul>
<li>add the heatmap on to a tree with or without annotations. See below for a tree with annotated clades and a heatmap.</li>
</ul>
<p>
</p>
<p><img src="reference/figures/Slide6.PNG"></p>

<p><strong>Known issues as October 23, 2020</strong></p>
<ul>
<li>If user uploads tree and genetic distance matrix and then annotates the tree, and then selects a meta data file, all previously placed annotations are wiped from the tree image.</li>
</ul>
<p>If you are interested in seeing the session info for the computer that developed this application please visit this <a href="https://github.com/jennahamlin/tinselR/issues/4">link</a>. Here all packages and versions that were installed on a windows machine are listed and successfully runs tinselR.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/jennahamlin/tinselR/">https://​github.com/​jennahamlin/​tinselR/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/jennahamlin/tinselR/issues">https://​github.com/​jennahamlin/​tinselR/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Jenna Hamlin <br><small class="roles"> Maintainer, author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Jenna Hamlin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
